src/main/java/io/kamax/matrix/client/MatrixHttpRoom.java

```diff
@@ -268,4 +274,120 @@ public _MatrixRoomMessageChunk getMessages(_MatrixRoomMessageChunkOptions option
         return new MatrixRoomMessageChunk(body.getStart(), body.getEnd(),
                 body.getChunk().stream().map(MatrixJsonPersistentEvent::new).collect(Collectors.toList()));
     }
+
+    @Override
+    public List<RoomTag> getUserTags() {
+        return getAllTags().stream().filter(tag -> "u".equals(tag.getNamespace())).collect(Collectors.toList());
+    }
+
+    private List<RoomTag> getAllTags() {
+        List<RoomTag> tags = new ArrayList<>();
+
+        URIBuilder builder = getClientPathBuilder("/user/{userId}/rooms/{roomId}/tags");
+
+        String body = execute(new HttpGet(getWithAccessToken(builder)));
+
+        if (StringUtils.isNotEmpty(body)) {
+            JsonObject jsonTags = jsonParser.parse(body).getAsJsonObject().get("tags").getAsJsonObject();
+            tags = jsonTags.entrySet().stream().filter(e -> e.getValue().isJsonObject()).map(entry -> {
```

Comment:
I'm not sure we should ignore invalid entries, if not an object. The body is wrong and not in line with the spec - so why not fail?

Reply:
To make sure I understand you correctly: Do you mean I should fail if `e.getValue()` is not a json object?

Reply:
Yes